File: android-native/KyouNoOgatore/app/src/main/java/jp/ogatore/kyouno/catalog/PlaylistData.kt
package jp.ogatore.kyouno.catalog

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

// playlists.json(scripts-native/gen-playlists.mjsがindex.html:3498-3521 PLAYLISTSから生成。
// TASK-C2-2026-07-28-search-playlists-and-fullwidth-space.md §1)のSerializableモデル。
// 手動キュレーションのYouTubeプレイリスト16本(3グループ)。CatalogData.ktと同じ方針。
@Serializable
data class PlaylistItem(
    val id: String,
    val title: String,
    val desc: String,
    val thumb: String? = null,
)

@Serializable
data class PlaylistGroup(
    val group: String,
    val items: List<PlaylistItem>,
)

object PlaylistLoader {
    val shared: List<PlaylistGroup> by lazy {
        val stream = PlaylistLoader::class.java.classLoader?.getResourceAsStream("playlists.json")
            ?: error("playlists.json がクラスパスに見つからない(app/src/main/resourcesの同梱漏れ)")
        val text = stream.bufferedReader(Charsets.UTF_8).use { it.readText() }
        Json { ignoreUnknownKeys = true }.decodeFromString(text)
    }
}

File: ios-native/KyouNoOgatore/KyouNoOgatore/PlaylistData.swift
//
//  PlaylistData.swift
//  KyouNoOgatore
//
//  playlists.json(scripts-native/gen-playlists.mjsがindex.html:3498-3521 PLAYLISTSから生成。
//  TASK-C2-2026-07-28-search-playlists-and-fullwidth-space.md §1)のDecodableモデル。
//  手動キュレーションのYouTubeプレイリスト16本(3グループ)。CatalogData.swiftと同じ方針。

import Foundation

struct PlaylistItem: Decodable {
    let id: String
    let title: String
    let desc: String
    let thumb: String?
}

struct PlaylistGroup: Decodable {
    let group: String
    let items: [PlaylistItem]
}

enum PlaylistLoader {
    static let shared: [PlaylistGroup] = {
        guard let url = Bundle.main.url(forResource: "playlists", withExtension: "json") else {
            fatalError("playlists.json がアプリバンドルに同梱されていない")
        }
        do {
            return try JSONDecoder().decode([PlaylistGroup].self, from: try Data(contentsOf: url))
        } catch {
            fatalError("playlists.json のデコードに失敗: \(error)")
        }
    }()
}
